airbyte-cdk/python/airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -128,7 +157,7 @@ def _validate_source(self):
         if "version" in self._source_config:
             full_config["version"] = self._source_config["version"]
         full_config["check"] = self._source_config["check"]
-        streams = [self._factory.create_component(stream_config, {}, False)() for stream_config in self._stream_configs()]
+        streams = [self._legacy_factory.create_component(stream_config, {}, False)() for stream_config in self._stream_configs()]
```

Comment:
Should we be flagging this on `self.construct_using_pydantic_models` so that we're validating the streams constructed by `self._constructor.create_component`? Or is blocked on implementing `create_declarative_stream` (in which case maybe a TODO here would be warranted?)?

Reply:
This is just a no-op for renaming the variable. I don't think we need to gate the validation behind anything because in the first part of the refactor we decided to validate the schema against both the auto-generated one (which is this block of code) as well as against the YAML handwritten one.  This should already be working as is, and the pydantic parsing we're doing in this PR is done after validations as we actually instantiate components.